Modules/Core/Common/include/itkMacro.h

```diff
@@ -115,20 +115,18 @@ namespace itk
 //
 // These macros respectively push and pop the diagnostic context
 //
-// Define macro ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP
 #if defined( __GNUC__ ) && !defined( __INTEL_COMPILER )
   #define ITK_GCC_PRAGMA_DIAG(x) ITK_PRAGMA(GCC diagnostic x)
   #define ITK_GCC_PRAGMA_DIAG_PUSH() ITK_GCC_PRAGMA_DIAG(push)
   #define ITK_GCC_PRAGMA_DIAG_POP() ITK_GCC_PRAGMA_DIAG(pop)
-  #define ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP
 #else
   #define ITK_GCC_PRAGMA_DIAG(x)
   #define ITK_GCC_PRAGMA_DIAG_PUSH()
   #define ITK_GCC_PRAGMA_DIAG_POP()
 #endif
 
 /*
- * ITK only supports MSVC++ 10.0 and greater
+ * ITK only supports MSVC++ 14.0 and greater
```

Comment:
We could replace the below table of versions by this updated and supported list:

MSVC++ 14.0  _MSC_VER == 1900 (Visual Studio 2015 version 14.0)
MSVC++ 14.1  _MSC_VER == 1910 (Visual Studio 2017 version 15.0)
MSVC++ 14.11 _MSC_VER == 1911 (Visual Studio 2017 version 15.3)
MSVC++ 14.12 _MSC_VER == 1912 (Visual Studio 2017 version 15.5)
MSVC++ 14.13 _MSC_VER == 1913 (Visual Studio 2017 version 15.6)
MSVC++ 14.14 _MSC_VER == 1914 (Visual Studio 2017 version 15.7)
MSVC++ 14.15 _MSC_VER == 1915 (Visual Studio 2017 version 15.8)
MSVC++ 14.16 _MSC_VER == 1916 (Visual Studio 2017 version 15.9)
MSVC++ 14.2  _MSC_VER == 1920 (Visual Studio 2019 Version 16.0)